lib/lease-manager.js

```diff
@@ -33,45 +56,48 @@ class LeaseManager extends EventEmitter {
    */
   initialize() {
     return this.provider.initialize().then((result) => {
-      this.status = STATUS.READY;
       this.data = result.data;
       this.lease_duration = result.lease_duration;
-      this.emit('ready');
-      this.error = null;
+
       this._renew();
+      if (this.status !== STATUS.READY) {
+        this.emit('ready');
+      }
+
       return this;
     }).catch((err) => {
-      this.status = STATUS.PENDING;
-      this.data = null;
-      this.lease_duration = 0;
-      if (this.listeners('error').length > 0) {
-        this.emit('error', err);
-      }
-      this.error = err;
+      this.emit('error', err);
+
       throw err;
     });
   }
 
   /**
    * Renew the lease from the SecretProvider
+   * @return {LeaseManager}
    * @private
    */
   _renew() {
     const timeout = (this.lease_duration / 2) * 1000; // eslint-disable-line rapid7/static-magic-numbers
 
-    setTimeout(() => {
+    if (this._timer && this._timeout === timeout) {
+      return this;
+    } else {
+      clearInterval(this._timer);
+      this._timeout = timeout;
+    }
+
+    this._timer = setInterval(() => {
       this.provider.renew().then((result) => {
-        this.lease_duration = result.lease_duration;
-        this.data = result.data;
-        this.emit('renewed');
+        this.emit('renewed', result);
         this._renew();
       }).catch((err) => {
-        this.error = err;
-        this.status = STATUS.PENDING;
-        this.lease_duration = 0;
+        this.emit('error', err);
```

Comment:
Previously, `this.data` wasn't being cleared if we failed to renew, because there's some period of time where the secret is still valid. I think we want to keep the behavior of only clearing `this.data` if the `initialize()` call fails.